WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalConnectionFailedUpdateAddress.swift

```diff
@@ -0,0 +1,65 @@
+import UIKit
+import Yosemite
+
+/// Modal presented when an error occurs while connecting to a reader due to problems with the address
+///
+final class CardPresentModalConnectingFailedUpdateAddress: CardPresentPaymentsModalViewModel {
+    private let continueSearchAction: () -> Void
+    private let cancelSearchAction: () -> Void
+
+    let textMode: PaymentsModalTextMode = .reducedTopInfo
+    let actionsMode: PaymentsModalActionsMode = .twoAction
+
+    let topTitle: String = Localization.title
+
+    var topSubtitle: String? = nil
+
+    let image: UIImage = .paymentErrorImage
+
+    let primaryButtonTitle: String? = Localization.retry
+
+    let secondaryButtonTitle: String? = Localization.cancel
+
+    let auxiliaryButtonTitle: String? = nil
+
+    var bottomTitle: String? = nil
+
+    let bottomSubtitle: String? = nil
+
+    init(continueSearch: @escaping () -> Void, cancelSearch: @escaping () -> Void) {
+        self.continueSearchAction = continueSearch
+        self.cancelSearchAction = cancelSearch
+    }
+
+    func didTapPrimaryButton(in viewController: UIViewController?) {
+        continueSearchAction()
+    }
+
+    func didTapSecondaryButton(in viewController: UIViewController?) {
+        cancelSearchAction()
+    }
+
+    func didTapAuxiliaryButton(in viewController: UIViewController?) { }
+}
+
+private extension CardPresentModalConnectingFailedUpdateAddress {
+    enum Localization {
+        static let title = NSLocalizedString(
+            "Please update your store address to proceed.",
```

Comment:
Could we be more specific? i.e. "update" seems vague. Could we use the word "correct" or "fully set" or something instead? cc @adamzelinski